kuma/settings/common.py

```diff
@@ -406,32 +406,33 @@ def enable_candidate_languages():
 )
 
 MIDDLEWARE_CLASSES = (
-    # must come before LocaleURLMiddleware
     'redirect_urls.middleware.RedirectsMiddleware',
-    # LocaleURLMiddleware must be before any middleware that uses
-    # kuma.core.urlresolvers.reverse() to add locale prefixes to URLs:
     'kuma.core.middleware.SetRemoteAddrFromForwardedFor',
     ('kuma.core.middleware.ForceAnonymousSessionMiddleware'
      if MAINTENANCE_MODE else
      'django.contrib.sessions.middleware.SessionMiddleware'),
+
+    # LocaleURLMiddleware requires access to request.user. These two must be
+    # loaded before the LocaleURLMiddleware
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'kuma.core.anonymous.AnonymousIdentityMiddleware',
+
+    # LocaleURLMiddleware must be before any middleware that uses
+    # kuma.core.urlresolvers.reverse() to add locale prefixes to URLs:
     'kuma.core.middleware.LocaleURLMiddleware',
     'kuma.wiki.middleware.DocumentZoneMiddleware',
     'kuma.wiki.middleware.ReadOnlyMiddleware',
     'kuma.core.middleware.Forbidden403Middleware',
     'django.middleware.common.CommonMiddleware',
     'kuma.core.middleware.RemoveSlashMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
-)
 
-if not MAINTENANCE_MODE:
     # We don't want this in maintence mode, as it adds "Cookie"
     # to the Vary header, which in turn, kills caching.
-    MIDDLEWARE_CLASSES += ('django.middleware.csrf.CsrfViewMiddleware',)
+    ('django.middleware.csrf.CsrfViewMiddleware'
+     if not MAINTENANCE_MODE else ''),
```

Comment:
I didn't test this yet, but I'll have to. I'm not 100% sure this will be valid when ``MAINTENANCE_MODE=True``.